docs/data/data-grid/recipes/MultilineEditing.tsx

```diff
@@ -0,0 +1,145 @@
+import * as React from 'react';
+import {
+  DataGridPro,
+  GridColDef,
+  GridRowModel,
+  GridRenderEditCellParams,
+  useGridApiContext,
+  GridColTypeDef,
+} from '@mui/x-data-grid-pro';
+import InputBase from '@mui/material/InputBase';
+import Popper from '@mui/material/Popper';
+import Paper from '@mui/material/Paper';
+import {
+  randomInt,
+  randomUserName,
+  randomArrayItem,
+} from '@mui/x-data-grid-generator';
+
+const lines = [
+  'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
+  'Aliquam dapibus, lorem vel mattis aliquet, purus lorem tincidunt mauris, in blandit quam risus sed ipsum.',
+  'Maecenas non felis venenatis, porta velit quis, consectetur elit.',
+  'Vestibulum commodo et odio a laoreet.',
+  'Nullam cursus tincidunt auctor.',
+  'Sed feugiat venenatis nulla, sit amet dictum nulla convallis sit amet.',
+  'Nulla venenatis justo non felis vulputate, eu mollis metus ornare.',
+  'Nam ullamcorper ligula id consectetur auctor.',
+  'Phasellus et ultrices dui.',
+  'Fusce facilisis egestas massa, et eleifend magna imperdiet et.',
+  'Pellentesque ac metus velit.',
+  'Vestibulum in massa nibh.',
+  'Vestibulum pulvinar aliquam turpis, ac faucibus risus varius a.',
+];
+
+const EditTextarea = (props: GridRenderEditCellParams<string>) => {
+  const { id, field, value, colDef, api } = props;
+  const [valueState, setValueState] = React.useState(value);
+  const [anchorEl, setAnchorEl] = React.useState<HTMLDivElement | null>();
+  const apiRef = useGridApiContext();
+
+  const handleRef = React.useCallback((el) => {
+    setAnchorEl(el);
+  }, []);
+
+  const handleChange = React.useCallback(
+    (event) => {
+      const newValue = event.target.value;
+      setValueState(newValue);
+      api.setEditCellValue({ id, field, value: newValue }, event);
+    },
+    [api, field, id],
+  );
+
+  const handleKeyDown = React.useCallback(
+    (event) => {
+      if (
+        event.key === 'Escape' ||
+        (event.key === 'Enter' &&
+          !event.shiftKey &&
+          (event.ctrlKey || event.metaKey))
+      ) {
+        console.log('publish events');
+        const params = apiRef.current.getCellParams(id, field);
+        apiRef.current.publishEvent('cellKeyDown', params, event);
+      }
```

Comment:
I assume that `apiRef.current.publishEvent('cellKeyDown', params, event);` would handle Escape properly.
My goal was to avoid replicating this logic and delegate it to `cellKeyDown` handler in the editing API (this demo uses new editing API)